src/openshift/application.ts

```diff
@@ -10,20 +10,36 @@ import * as validator from 'validator';
 
 export class Application extends OpenShiftItem {
     static async create(project: OpenShiftObject): Promise<String> {
-        const appName = await vscode.window.showInputBox({
-            prompt: "Application name",
-            validateInput: (value: string) => {
-                if (validator.isEmpty(value.trim())) {
-                    return 'Empty application name';
-                }
-                if (!validator.matches(value.trim(), '^[a-z0-9]([-a-z0-9]*[a-z0-9])*$')) {
-                    return 'Not a valid application name. Please use lower case alphanumeric characters or "-", and must start and end with an alphanumeric character';
-                }
+        let projectObj;
+        let appName;
+        if (project) {
+            projectObj = project;
+        } else {
+            projectObj = await Application.odo.getProjects();
+            if (projectObj.length === 0) {
+                await vscode.window.showErrorMessage('Sorry there are no project to create an application (Please create new project and try again)');
+            }else if (projectObj.length > 1 ) {
+                projectObj = await vscode.window.showQuickPick(projectObj, {placeHolder: "In which project you want to create Application"});
+            } else if (projectObj[0]) {
+                projectObj = projectObj[0];
             }
-        });
+        }
+        if (!(projectObj.length === 0)) {
```

Comment:
should be just 
``` javascript
if (projectObj) {
```
considering you not using `projectObj` just to store selected oproject